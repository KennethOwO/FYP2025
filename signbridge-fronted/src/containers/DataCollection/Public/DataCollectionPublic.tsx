import React from "react";
import DataCollection from "../components/DataCollection/DataCollection";
const DataCollectionPublic = () => {
  return (
    <div>
      <DataCollection user="public" />
    </div>
  );
};

export default DataCollectionPublic;
