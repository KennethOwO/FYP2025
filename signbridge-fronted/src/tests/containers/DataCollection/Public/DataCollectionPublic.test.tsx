import { create } from "react-test-renderer";
import {
  render,
  fireEvent,
  cleanup,
  screen,
  waitFor,
} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom"; //For extra validation => toBeInTheDocument()
import DataCollectionPublic from "../../../../containers/DataCollection/Public/DataCollectionPublic";
import { act } from "react-dom/test-utils";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: string) => key, // Mock the translation function to return the key
    i18n: {
      changeLanguage: jest.fn(), // Mock the changeLanguage function
    },
  }),
}));

// Mock the useSpeechRecognition hook
jest.mock("react-speech-recognition", () => ({
  useSpeechRecognition: () => ({
    transcript: "Hello, World!",
    resetTranscript: jest.fn(),
    browserSupportsSpeechRecognition: true,
  }),
}));

jest.mock("js-cookie");

describe("Test DataCollectionPublic", () => {
  afterEach(() => {
    // Clean up the DOM
    cleanup();
    // Reset mocks to their initial state
    jest.resetAllMocks();
  });
  it("should render correctly", () => {
    const tree = create(<DataCollectionPublic />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should be able to submit", async () => {
    // Mocking the selected video file => For video upload
    Object.defineProperty(global, "File", {
      value: class extends Blob {
        constructor(
          fileParts: Array<string>,
          fileName: string,
          options?: BlobPropertyBag
        ) {
          super(fileParts, options);
          (this as any).name = fileName;
        }
      },
    });

    // Use asFragment to render toMatchSnapshot
    const { asFragment } = render(<DataCollectionPublic />);

    // Fill in the form fields
    userEvent.type(screen.getByTestId("name"), "John Doe"); // If you go to my code you will find that I have put up testid for it, it's easier to use
    userEvent.type(screen.getByTestId("email"), "john@example.com");
    userEvent.type(screen.getByTestId("text"), "Test sentence");

    // Mock the Upload component behavior
    jest.mock("antd", () => ({
      Upload: ({ children }: { children: React.ReactNode }) => (
        <div data-testid="upload-mock">{children}</div>
      ),
    }));

    // Click the submit button
    await act(async () => {
      fireEvent.click(screen.getByTestId("click-close-btn"));
      fireEvent.click(screen.getByTestId("submit_btn2"));
    });

    // Using snapshot as temporary solution
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText("OK")).toBeTruthy();
  });
});
