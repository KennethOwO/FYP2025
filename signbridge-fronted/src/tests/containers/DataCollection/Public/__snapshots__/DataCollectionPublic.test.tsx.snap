// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DataCollectionPublic should be able to submit 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="dataForm-bg"
      >
        <div
          class="popup-confirmation "
        >
          <div
            class="popup-confirmation-content-header"
          >
            <h2>
              insAreYouSure
            </h2>
          </div>
          <div
            class="popup-confirmation-content"
          >
            <div
              class="popup-confirmation-details"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-175ib4q-MuiPaper-root-MuiTableContainer-root"
              >
                <table
                  class="MuiTable-root css-rqglhn-MuiTable-root"
                >
                  <tbody
                    class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                  >
                    <tr
                      class="MuiTableRow-root css-ys9ggv-MuiTableRow-root"
                    >
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
                      >
                        <b>
                          insName
                        </b>
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-3urw24-MuiTableCell-root"
                      />
                    </tr>
                    <tr
                      class="MuiTableRow-root css-2qi3sl-MuiTableRow-root"
                    >
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
                      >
                        <b>
                          insEmail
                        </b>
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1vhc2pi-MuiTableCell-root"
                      />
                    </tr>
                    <tr
                      class="MuiTableRow-root css-ys9ggv-MuiTableRow-root"
                    >
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
                      >
                        <b>
                          insText
                        </b>
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1vhc2pi-MuiTableCell-root"
                      >
                        Hello, World!
                      </td>
                    </tr>
                    <tr
                      class="MuiTableRow-root css-1yq2m0q-MuiTableRow-root"
                    >
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
                      >
                        <b>
                          insVideo
                        </b>
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1vhc2pi-MuiTableCell-root"
                      />
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
            <div
              class="popup-confirmation-content-button"
            >
              <button
                class="btn btn--reset btn--large"
                data-testid="button-btn--reset"
                type="button"
              >
                cancel_btn
              </button>
              <button
                class="ant-btn css-dev-only-do-not-override-42nv3w ant-btn-default ant-btn-loading btn-submit btn-large"
                data-testid="submit_btn2"
                style="height: auto; font-size: 20px; position: relative;"
                type="button"
              >
                <span
                  class="ant-btn-icon ant-btn-loading-icon"
                >
                  <span
                    aria-label="loading"
                    class="anticon anticon-loading anticon-spin"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="loading"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 1024 1024"
                      width="1em"
                    >
                      <path
                        d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                      />
                    </svg>
                  </span>
                </span>
                <span>
                  submit_btn
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="undefined "
        >
          <div>
            <h1>
              instructions
            </h1>
            <i
              class="undefined fa fa-close"
              data-testid="click-close-btn"
            />
          </div>
          <div>
            <div>
              <p>
                <strong>
                  1.
                </strong>
                 inslogin
              </p>
              <p>
                <strong>
                  2.
                </strong>
                 insFillDetails
              </p>
              <p>
                <strong>
                  3.
                </strong>
                 insUploadDemoVideo
              </p>
              <p>
                <strong>
                  4.
                </strong>
                 insResetBtn
              </p>
            </div>
            <hr />
            <div>
              <p>
                insNote
              </p>
              <div>
                <ul>
                  <li>
                    insVideoBg
                  </li>
                  <li>
                    insVideoMp4
                  </li>
                  <li>
                    insVideoBody
                  </li>
                  <li>
                    insVideoPurpose
                  </li>
                </ul>
              </div>
            </div>
            <div>
              <p>
                insReferImage
              </p>
              <div>
                <img
                  alt="instruction-image"
                  src="default"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="dataForm"
        >
          <div
            class="dataForm-header-container"
          >
            <div
              class="dataForm-header"
            >
              <h1>
                dataset_collection_form
              </h1>
              <p>
                datasetmsg
              </p>
            </div>
          </div>
          <div
            class="dataForm-cover"
          >
            <div
              class="dataForm-card public"
            >
              <h1>
                in_touch
              </h1>
              <h3>
                more_suggestions
              </h3>
              <div
                class="dataForm-card-content"
              >
                <div
                  class="dataForm-card-info"
                >
                  <svg
                    fill="currentColor"
                    height="2em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(0, 0, 0); margin-right: 5px;"
                    viewBox="0 0 24 24"
                    width="2em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 2C8.13 2 5 5.13 5 9c0 1.74.5 3.37 1.41 4.84.95 1.54 2.2 2.86 3.16 4.4.47.75.81 1.45 1.17 2.26.26.55.47 1.5 1.26 1.5s1-.95 1.25-1.5c.37-.81.7-1.51 1.17-2.26.96-1.53 2.21-2.85 3.16-4.4C18.5 12.37 19 10.74 19 9c0-3.87-3.13-7-7-7zm0 9.75a2.5 2.5 0 0 1 0-5 2.5 2.5 0 0 1 0 5z"
                    />
                  </svg>
                  <p>
                     neoun_address
                  </p>
                </div>
                <div
                  class="dataForm-card-info2"
                >
                  <svg
                    fill="currentColor"
                    height="2em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(0, 0, 0); margin-right: 5px;"
                    viewBox="0 0 24 24"
                    width="2em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
                    />
                  </svg>
                  <p>
                    neoun_phone
                  </p>
                </div>
                <div
                  class="dataForm-card-info2"
                >
                  <svg
                    fill="currentColor"
                    height="30px"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(0, 0, 0); margin-right: 5px;"
                    viewBox="0 0 24 24"
                    width="30px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                    />
                  </svg>
                  <p>
                    neoun_email
                  </p>
                </div>
                <div
                  class="dataForm-card-info"
                />
              </div>
            </div>
            <div
              class="dataForm-container"
            >
              <div
                class="dataForm-info-block"
              >
                <svg
                  class="dataForm-info"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: rgb(0, 0, 0);"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 4v16a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1V4a1 1 0 0 0-1-1H4a1 1 0 0 0-1 1zm8 3h2v2h-2V7zm0 4h2v6h-2v-6z"
                  />
                </svg>
              </div>
              <div
                class="row justify-content-center input-container"
              >
                <div
                  class="col-md-8"
                >
                  <div
                    class="form-group "
                  >
                    <input
                      class="form-control "
                      data-testid="name"
                      name="name"
                      placeholder=" "
                      type="text"
                      value=""
                    />
                    <label
                      class="input-label "
                    >
                      name_input
                    </label>
                    <div
                      class="form-error "
                    />
                  </div>
                  <div
                    class="form-group "
                  >
                    <input
                      class="form-control "
                      data-testid="email"
                      name="email"
                      placeholder=" "
                      type="email"
                      value=""
                    />
                    <label
                      class="input-label "
                    >
                      email_input
                    </label>
                    <div
                      class="form-error "
                    />
                  </div>
                  <div
                    class="voiceToTextBox"
                  >
                    <div
                      class="form-group "
                    >
                      <textarea
                        class="form-control  multipleLines"
                        data-testid="text"
                        name="text"
                        placeholder=" "
                      >
                        Hello, World!
                      </textarea>
                      <label
                        class="input-label multiple_label"
                      >
                        text_sentence_input
                      </label>
                      <div
                        class="form-error "
                      />
                    </div>
                    <button
                      class="avatar-microphone-btn"
                    >
                      <i
                        class="fa fa-microphone"
                      />
                    </button>
                     
                  </div>
                  <div
                    class="video-container"
                  >
                    <div
                      class="videoinput-class"
                    >
                      <span
                        class="ant-upload-wrapper css-dev-only-do-not-override-42nv3w"
                      >
                        <div
                          class="ant-upload ant-upload-select"
                        >
                          <span
                            class="ant-upload"
                          >
                            <input
                              accept=".mp4"
                              style="display: none;"
                              type="file"
                            />
                            <button
                              class="ant-btn css-dev-only-do-not-override-42nv3w ant-btn-default ant-btn-lg"
                              type="button"
                            >
                              <span
                                class="ant-btn-icon"
                              >
                                <span
                                  aria-label="upload"
                                  class="anticon anticon-upload"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="upload"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                                    />
                                  </svg>
                                </span>
                              </span>
                              <span>
                                choose_a_video
                              </span>
                            </button>
                          </span>
                        </div>
                      </span>
                    </div>
                  </div>
                  <div
                    class="button-container"
                  >
                    <button
                      class="btn btn--reset btn--large"
                      data-testid="button-btn--reset"
                      type="button"
                    >
                      reset_btn
                    </button>
                    <button
                      class="btn btn--submit btn--large"
                      data-testid="button-btn--submit"
                      type="submit"
                    >
                      submit_btn
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="popup-modal  modal-dialog-centered modal-lg"
      >
        <img
          alt=""
          class="popup-tick img-fluid"
          src="default"
        />
        <div
          class="modal-content"
        >
          <h2
            class="modal-title"
          >
            insThankYou
          </h2>
          <p
            class="modal-body"
          >
            insProceedDetails
          </p>
          <button
            class="btn btn-success"
          >
            OK
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test DataCollectionPublic should render correctly 1`] = `
<div>
  <div>
    <div
      className="dataForm-bg"
    >
      <div
        className="popup-confirmation "
      >
        <div
          className="popup-confirmation-content-header"
        >
          <h2>
            insAreYouSure
          </h2>
        </div>
        <div
          className="popup-confirmation-content"
        >
          <div
            className="popup-confirmation-details"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-175ib4q-MuiPaper-root-MuiTableContainer-root"
            >
              <table
                className="MuiTable-root css-rqglhn-MuiTable-root"
                role={null}
              >
                <tbody
                  className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                  role={null}
                >
                  <tr
                    className="MuiTableRow-root css-ys9ggv-MuiTableRow-root"
                    role={null}
                  >
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
                    >
                      <b>
                        insName
                      </b>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-3urw24-MuiTableCell-root"
                    />
                  </tr>
                  <tr
                    className="MuiTableRow-root css-2qi3sl-MuiTableRow-root"
                    role={null}
                  >
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
                    >
                      <b>
                        insEmail
                      </b>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1vhc2pi-MuiTableCell-root"
                    />
                  </tr>
                  <tr
                    className="MuiTableRow-root css-ys9ggv-MuiTableRow-root"
                    role={null}
                  >
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
                    >
                      <b>
                        insText
                      </b>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1vhc2pi-MuiTableCell-root"
                    >
                      Hello, World!
                    </td>
                  </tr>
                  <tr
                    className="MuiTableRow-root css-1yq2m0q-MuiTableRow-root"
                    role={null}
                  >
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-r6fuq3-MuiTableCell-root"
                    >
                      <b>
                        insVideo
                      </b>
                    </td>
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1vhc2pi-MuiTableCell-root"
                    />
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
          <div
            className="popup-confirmation-content-button"
          >
            <button
              className="btn btn--reset btn--large"
              data-testid="button-btn--reset"
              onClick={[Function]}
              type="button"
            >
              cancel_btn
            </button>
            <button
              className="ant-btn css-dev-only-do-not-override-42nv3w ant-btn-default btn-submit btn-large"
              data-testid="submit_btn2"
              disabled={false}
              onClick={[Function]}
              style={
                {
                  "fontSize": "20px",
                  "height": "auto",
                  "position": "relative",
                }
              }
              type="button"
            >
              <span>
                submit_btn
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        className="undefined "
      >
        <div>
          <h1>
            instructions
          </h1>
          <i
            className="undefined fa fa-close"
            data-testid="click-close-btn"
            onClick={[Function]}
          />
        </div>
        <div>
          <div>
            <p>
              <strong>
                1.
              </strong>
               
              inslogin
            </p>
            <p>
              <strong>
                2.
              </strong>
               
              insFillDetails
            </p>
            <p>
              <strong>
                3.
              </strong>
               
              insUploadDemoVideo
            </p>
            <p>
              <strong>
                4.
              </strong>
               
              insResetBtn
            </p>
          </div>
          <hr />
          <div>
            <p>
              insNote
            </p>
            <div>
              <ul>
                <li>
                  insVideoBg
                </li>
                <li>
                  insVideoMp4
                </li>
                <li>
                  insVideoBody
                </li>
                <li>
                  insVideoPurpose
                </li>
              </ul>
            </div>
          </div>
          <div>
            <p>
              insReferImage
            </p>
            <div>
              <img
                alt="instruction-image"
                src="default"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="dataForm"
      >
        <div
          className="dataForm-header-container"
        >
          <div
            className="dataForm-header"
          >
            <h1>
              dataset_collection_form
            </h1>
            <p>
              datasetmsg
            </p>
          </div>
        </div>
        <div
          className="dataForm-cover"
        >
          <div
            className="dataForm-card public"
          >
            <h1>
              in_touch
            </h1>
            <h3>
              more_suggestions
            </h3>
            <div
              className="dataForm-card-content"
            >
              <div
                className="dataForm-card-info"
              >
                <svg
                  fill="currentColor"
                  height="2em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    {
                      "color": "#000",
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 24 24"
                  width="2em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12 2C8.13 2 5 5.13 5 9c0 1.74.5 3.37 1.41 4.84.95 1.54 2.2 2.86 3.16 4.4.47.75.81 1.45 1.17 2.26.26.55.47 1.5 1.26 1.5s1-.95 1.25-1.5c.37-.81.7-1.51 1.17-2.26.96-1.53 2.21-2.85 3.16-4.4C18.5 12.37 19 10.74 19 9c0-3.87-3.13-7-7-7zm0 9.75a2.5 2.5 0 0 1 0-5 2.5 2.5 0 0 1 0 5z"
                  />
                </svg>
                <p>
                   
                  neoun_address
                </p>
              </div>
              <div
                className="dataForm-card-info2"
              >
                <svg
                  fill="currentColor"
                  height="2em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    {
                      "color": "#000",
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 24 24"
                  width="2em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
                  />
                </svg>
                <p>
                  neoun_phone
                </p>
              </div>
              <div
                className="dataForm-card-info2"
              >
                <svg
                  fill="currentColor"
                  height="30px"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    {
                      "color": "#000",
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 24 24"
                  width="30px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                  />
                </svg>
                <p>
                  neoun_email
                </p>
              </div>
              <div
                className="dataForm-card-info"
              />
            </div>
          </div>
          <div
            className="dataForm-container"
          >
            <div
              className="dataForm-info-block"
            >
              <svg
                className="dataForm-info"
                fill="currentColor"
                height="1em"
                onClick={[Function]}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": "#000",
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 4v16a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1V4a1 1 0 0 0-1-1H4a1 1 0 0 0-1 1zm8 3h2v2h-2V7zm0 4h2v6h-2v-6z"
                />
              </svg>
            </div>
            <div
              className="row justify-content-center input-container"
            >
              <div
                className="col-md-8"
              >
                <div
                  className="form-group "
                >
                  <input
                    className="form-control "
                    data-testid="name"
                    name="name"
                    onChange={[Function]}
                    placeholder=" "
                    type="text"
                    value=""
                  />
                  <label
                    className="input-label "
                  >
                    name_input
                  </label>
                  <div
                    className="form-error "
                  />
                </div>
                <div
                  className="form-group "
                >
                  <input
                    className="form-control "
                    data-testid="email"
                    name="email"
                    onChange={[Function]}
                    placeholder=" "
                    type="email"
                    value=""
                  />
                  <label
                    className="input-label "
                  >
                    email_input
                  </label>
                  <div
                    className="form-error "
                  />
                </div>
                <div
                  className="voiceToTextBox"
                >
                  <div
                    className="form-group "
                  >
                    <textarea
                      className="form-control  multipleLines"
                      data-testid="text"
                      name="text"
                      onChange={[Function]}
                      placeholder=" "
                      value="Hello, World!"
                    />
                    <label
                      className="input-label multiple_label"
                    >
                      text_sentence_input
                    </label>
                    <div
                      className="form-error "
                    />
                  </div>
                  <button
                    className="avatar-microphone-btn"
                    onClick={[Function]}
                  >
                    <i
                      className="fa fa-microphone"
                    />
                  </button>
                   
                </div>
                <div
                  className="video-container"
                >
                  <div
                    className="videoinput-class"
                  >
                    <span
                      className="ant-upload-wrapper css-dev-only-do-not-override-42nv3w"
                    >
                      <div
                        className="ant-upload ant-upload-select"
                      >
                        <span
                          className="ant-upload"
                          onClick={[Function]}
                          onDragOver={[Function]}
                          onDrop={[Function]}
                          onKeyDown={[Function]}
                        >
                          <input
                            accept=".mp4"
                            disabled={false}
                            multiple={false}
                            onChange={[Function]}
                            onClick={[Function]}
                            style={
                              {
                                "display": "none",
                              }
                            }
                            type="file"
                          />
                          <button
                            className="ant-btn css-dev-only-do-not-override-42nv3w ant-btn-default ant-btn-lg"
                            disabled={false}
                            onClick={[Function]}
                            style={{}}
                            type="button"
                          >
                            <span
                              className="ant-btn-icon"
                              style={{}}
                            >
                              <span
                                aria-label="upload"
                                className="anticon anticon-upload"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="upload"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                                  />
                                </svg>
                              </span>
                            </span>
                            <span>
                              choose_a_video
                            </span>
                          </button>
                        </span>
                      </div>
                    </span>
                  </div>
                </div>
                <div
                  className="button-container"
                >
                  <button
                    className="btn btn--reset btn--large"
                    data-testid="button-btn--reset"
                    onClick={[Function]}
                    type="button"
                  >
                    reset_btn
                  </button>
                  <button
                    className="btn btn--submit btn--large"
                    data-testid="button-btn--submit"
                    onClick={[Function]}
                    type="submit"
                  >
                    submit_btn
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="popup-modal  modal-dialog-centered modal-lg"
    >
      <img
        alt=""
        className="popup-tick img-fluid"
        src="default"
      />
      <div
        className="modal-content"
      >
        <h2
          className="modal-title"
        >
          insThankYou
        </h2>
        <p
          className="modal-body"
        >
          insProceedDetails
        </p>
        <button
          className="btn btn-success"
          onClick={[Function]}
        >
          OK
        </button>
      </div>
    </div>
  </div>
</div>
`;
